spec/tbd_tests_spec.rb

```diff
@@ -1591,4 +1591,261 @@
+        expect(sub.include?(side))
+        expect(sub.include?(" Window"))
+      end
+
+      # 2nd tallies, per office ouside-facing wall:
+      #   - nb: number of distinct edges, per MAJOR thermal bridge type
+      #   - lm: total edge lengths (m), per MAJOR thermal bridge type
+      jambs2   = { nb: 0, lm: 0 }
+      sills2   = { nb: 0, lm: 0 }
+      heads2   = { nb: 0, lm: 0 }
+      grades2  = { nb: 0, lm: 0 }
+      rims2    = { nb: 0, lm: 0 }
+      corners2 = { nb: 0, lm: 0 }
+
+      io[:edges].each do |edge|
+        expect(edge.key?(:surfaces))
+        expect(edge[:surfaces].is_a?(Array))
+        expect(edge[:surfaces].empty?).to be(false)
+        next unless edge[:surfaces].include?(side)
+
+        expect(edge.key?(:length))
+        expect(edge.key?(:type  ))
+        next if edge[:type] == :transition
+
+        case edge[:type]
+        when :jamb
+          jambs2[:nb] += 1
+          jambs2[:lm] += edge[:length]
+        when :sill
+          sills2[:nb] += 1
+          sills2[:lm] += edge[:length]
+        when :head
+          heads2[:nb] += 1
+          heads2[:lm] += edge[:length]
+        when :gradeconvex
+          grades2[:nb] += 1
+          grades2[:lm] += edge[:length]
+        when :rimjoist
+          rims2[:nb] += 1
+          rims2[:lm] += edge[:length]
+        else
+          corners2[:nb] += 1
+          corners2[:lm] += edge[:length]
+        end
+      end
+
+      expect(  jambs2[:nb]).to eq(6) # no change vs initial, unaltered model
+      expect(  sills2[:nb]).to eq(2)
+      expect(  heads2[:nb]).to eq(3)
+      expect( grades2[:nb]).to eq(3)
+      expect(   rims2[:nb]).to eq(1)
+      expect(corners2[:nb]).to eq(1)
+
+      if side == front
+        expect(  jambs2[:lm]).to be_within(0.01).of(10.37 * mult)
+        expect(  sills2[:lm]).to be_within(0.01).of( 7.31 * mult)
+        expect(  heads2[:lm]).to be_within(0.01).of( 9.14 * mult)
+        expect(   rims2[:lm]).to be_within(0.01).of(25.91) # unchanged
+        expect(corners2[:lm]).to be_within(0.01).of( 4.27) # unchanged
+
+        # In the OpenStudio warehouse model, the front door (2x 915mm) "sill" is
+        # aligned along the slab-on-"grade" edge. It is the only such "shared"
+        # subsurface edge in this (front wall) example. In TBD, such common
+        # edges initially hold multiple thermal bridge types in memory, until a
+        # single, dominant type (based on PSI factor) is finally assigned
+        # (here, "grade" not "sill").
+        #
+        # By adding a 2x multiplier in OpenStudio, door area and perimeter have
+        # doubled while the initial subsurface vertices remain as before. This
+        # of course breaks model geometrical consistency (vs Topolys), which is
+        # expected with multipliers - they remain abstract modifiers that do not
+        # lend easily to 3D representation. It would be imprudent for
+        # TBD/Topolys to "stretch" subsurface vertex 3D position, based on
```

Comment:
Agree that you don't want to change the Topolys model.  Seems like you want to convert the quasi-physical Topolys model to a thermal model and then do all the multiplier and other adjustments on that.